Sources/Testing/Running/Configuration.swift

```diff
@@ -118,38 +118,49 @@ public struct Configuration: Sendable {
   /// - Returns: A Boolean value representing if the test satisfied the filter.
   public typealias TestFilter = @Sendable (Test) -> Bool
 
+  /// Storage for ``testFilter``.
+  private var _testFilter: TestFilter = { !$0.isHidden }
+
   /// The test filter to which tests should be filtered when run.
-  public var testFilter: TestFilter?
+  public var testFilter: TestFilter {
+    get {
+      _testFilter
+    }
+    set {
+      // By default, the test filter should always filter out hidden tests. This
+      // is the appropriate behavior for external clients of this SPI. If the
+      // testing library needs to enable hidden tests in its own test targets,
+      // it should use setTestFilter(toMatch:includeHiddenTests:) instead.
+      _testFilter = { test in
+        !test.isHidden && newValue(test)
+      }
+    }
+  }
 
   /// Filter tests to run to those specified via a set of test IDs.
   ///
   /// - Parameters:
-  ///   - selection: A set of test IDs to be filtered. If `nil`, the current
-  ///     selection is cleared.
+  ///   - selection: A set of test IDs to be filtered.
   ///
   /// By default, all tests are run and no filter is set.
-  public mutating func setTestFilter(toMatch selection: Set<Test.ID>?) {
-    setTestFilter(toMatch: selection.map(Test.ID.Selection.init), includeHiddenTests: false)
+  public mutating func setTestFilter(toMatch selection: Set<Test.ID>) {
```

Comment:
By making the argument here non-optional, we simplify the logic quite a bit. If a caller wants to reset the test filter, they can set it to `{ _ in true }`, but in practice I don't think mutating a test filter in-flight is a concern for us.

Reply:
okay!